### DISCUSSION TITLE
vsg::View deleting in runtime

### DISCUSSION THREAD

**Discussion by Anshi701** [vsg-dev/VulkanSceneGraph#1070] [Q&A]
Let's say i started some vsg app and created View. When i want to delete this view. I'm doing it with the next steps:
 - All operations are going in Update stage of rendering.
 - I'm finding and deleting View node from renderGraph of window with simple remove function of vector.
 - Reset View object.
 - Compile renderGraph with compileManager of Viewer.
 - Update Viewer with compile result.
After this steps, no View is displaying in window. But then i am creating new View object on different location of the screen, assign ClearAttachments and new ViewportState. And there is a problem - ClearAttachments is on the right place, but actual render is on location of previous View. 
![image](https://github.com/vsg-dev/VulkanSceneGraph/assets/88488743/d782c57f-653b-474a-aac3-de56513a43a2)
Is VSG support View deleting in runtime, or what i am doing wrong?

**Comment by robertosfield** [comment]
What is likely happening is the scene graph's GraphicsPipeline has built a vkPipeline for the original View's viewport dimensions but you've got rid of it and replaced it with a new View with new viewport dimensions but without rebuild the vkPipeline objects.   

The vsgmultiviews example doesn't do exactly what you want but it does have some code that swaps the ViewportState and updates the pipelines so perhaps this might be a help if figuring out exactly what you need to do in your usage case:

https://github.com/vsg-dev/vsgExamples/blob/master/examples/app/vsgmultiviews/vsgmultiviews.cpp#L39

**Reply by Anshi701** [reply]
Thanks for fast reply! I will check it out and share with results.
